powerquery-docs/dataflows/dataflow-solution-awareness.md

```diff
@@ -92,6 +92,7 @@ For security reasons, credentials of connections used by dataflows aren't persis
 * Dataflows can't use connection references for any connector.
 * Environment variables can't be used by dataflows.
 * Dataflows don't support adding required components, such as custom tables they load data to. Instead, the custom table should be manually added to the solution.
-* Dataflows can't be deployed by application users (service principals).
+* Dataflows can't be deployed by application users (service principals). The solution will import, but the imported dataflows can't be edited or reassigned.
+* Dataflows deployed by a guest user can't be reassigned. Only the guest user importing the solution will be able to edit the dataflow.
```

Comment:
Hi @Kunter-Bunt, that you for suggesting this change!

You're right that guest users are filtered out from selection picker. I checked for historical reason, and it was done to filter out external users. 

Do you mind elaborating what use you had which brought you to making this documentation change? Do you have guest users that are original users of your tenant? (i.e. identities created in the tenant being discussed and not B2B invited from a different tenant)

Reply:
Hi @msftyehong,
I didn't have a use case here, I ran into this when I imported a solution with a Dataflow using an App Registration. 
While I can't create Dataflows with an AppReg, I can import solutions and by definition any new object created by solution import is owned by the user importing. Thus, the Dataflow ended up being owned by the AppReg.
While trying to reassign, I found the query and figured this would also be true for guest users: I have double checked this now. While I can't create a dataflow due to some 500 error in the connections API, I can of course import a solution with a Dataflow and thus own it with the guest user. 

![image](https://github.com/MicrosoftDocs/powerquery-docs/assets/9168648/06a18e49-8b5a-4aa5-8c21-736c0bb202f6)
![2024-01-29_20h54_16](https://github.com/MicrosoftDocs/powerquery-docs/assets/9168648/d9743bf2-5718-460d-8eda-71dc9f4761f7)


I further checked what happens if I manipulate the JS in this case: If I lift this limitation, I can reassign the Dataflow without any problems. I can also assign a Dataflow owned by a member to a guest (if I manipulate the search query).
Not to say that this limitation should simply be removed as Dataflows are apparently not tested for guest users, but I figured that the inability to find the guest user and reassign the Dataflow should be documented. Importing a solution with a Dataflow as a guest will leave you in a bad state if the "Dataflow Maker Experience" is not working to set the connection or the guest is being deactivated or a reassignment needs to happen for any other reason.

![image](https://github.com/MicrosoftDocs/powerquery-docs/assets/9168648/46e9b559-e178-4b3f-b9ce-82256f082529)

Reply:
Thanks for the feedback. 
To not guarantee any behavior I also changed the wording on guest users here. If of course the limitation can be lifted, the added line is obsolete.

Another thought on wording: I initially added the additional sentence because "can't" is obviously not true. Maybe something like 
"_Dataflows must not be deployed by application users (service principals)._" or "_Deploying dataflows with application users (service principals) is not supported._" better describes the actual situation.

Reply:
Sorry for the long time it has taken us to get back to you. We found that the fix is simple, so we took it. You should be able to review a guest user Dataflows and change their owner. I've confirmed in internal test environment - if you cannot see it at your region, please test it again after 03/16 - when it should be deployed everywhere.

Thank you for reporting this issue by submitting this limitation change!